examples/quickrefl/core/editors/sldeditor/viewobjects/elementview.cpp

```diff
@@ -0,0 +1,149 @@
+// ************************************************************************** //
+//
+//  Model-view-view-model framework for large GUI applications
+//
+//! @license   GNU General Public License v3 or higher (see COPYING)
+//! @authors   see AUTHORS
+//
+// ************************************************************************** //
+
+#include "elementview.h"
+#include "GraphicsScene.h"
+#include "mvvm/plotting/sceneadapterinterface.h"
+
+#include <QPainter>
+#include <QPainterPath>
+
+//! The constructor
+ElementView::ElementView() : QGraphicsObject() {}
+ElementView::~ElementView() = default;
+
+//! Get the conversion axes
+ModelView::SceneAdapterInterface* ElementView::getSceneAdapter() const
+{
+    GraphicsScene* scene_item = static_cast<GraphicsScene*>(scene());
+    if (!scene_item)
+        return nullptr;
+
+    return scene_item->getSceneAdapter();
+}
+
+//! Advance method used by the scene adapter
+void ElementView::advance(int phase)
+{
+    if (!phase)
+        return;
+    prepareGeometryChange();
+    update();
+}
+
+//! paint override
+void ElementView::paint(QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget)
+{
+    painter->setClipRect(getSceneAdapter()->viewportRectangle());
+}
+
+//! modify the rectangle for display according to the scene adapter
+QRectF ElementView::displayRect(QRectF real_rect) const
+{
+    auto adapter = getSceneAdapter();
+    if (!adapter)
+        return real_rect;
+
+    double x = real_rect.x();
+    double y = real_rect.y();
+    double w = real_rect.width();
+    double h = real_rect.height();
+
+    if (m_center_based) {
+        double center_x = x + w / 2.;
+        double center_y = y + h / 2.;
+
+        if (m_adapt_x) {
+            center_x = adapter->toSceneX(center_x);
+        }
+        if (m_adapt_y) {
+            center_y = adapter->toSceneY(center_y);
+        }
+        if (m_adapt_width) {
+            w = adapter->toSceneX(w) - adapter->toSceneX(0);
+        }
+        if (m_adapt_height) {
+            h = adapter->toSceneY(h) - adapter->toSceneY(0);
+        }
+
+        x = center_x - w / 2;
+        y = center_y - h / 2;
+
+    } else {
+        if (m_adapt_x) {
```

Comment:
What is the purpose of `m_adapt_x` flags?

Reply:
Some ViewItem properties should be adapted through the help of the scene adapter while others should not. This assures, for example, that the handle has always the same size on the scene when zooming in and out. Therefore the scene adapter only changes property that should scale among x, y, width (w) and the height (h).

Reply:
ok, tx